### ISSUE TITLE
New-AzEventGridSubscription has different parameters on cloud shell and on local machine with the same version.

### ISSUE THREAD

**Issue by bandolir** [Azure/azure-powershell#19306]
### Description

Despite of I have the same version or newer from each necessary modules the windows version of New-AzEventGridSubscription has less parameters than Linux version. e.g. DeliveryAttributeMapping is missing.

Full parameters list
[CloudShell-AzEventGridSubscriptionOutput.txt](https://github.com/Azure/azure-powershell/files/9414534/CloudShell-AzEventGridSubscriptionOutput.txt)
[Local-New-AzEventGridSubscrptionOutput.txt](https://github.com/Azure/azure-powershell/files/9414538/Local-New-AzEventGridSubscrptionOutput.txt)

Modules are same and there is no other modules folder on my local machine.
[CloudShell-GetModuleOutput.txt](https://github.com/Azure/azure-powershell/files/9414535/CloudShell-GetModuleOutput.txt)
[Local-GetModuleOutput.txt](https://github.com/Azure/azure-powershell/files/9414537/Local-GetModuleOutput.txt)

The only difference I see is the OS/Platform.
[CloudShell-VersionTable.txt](https://github.com/Azure/azure-powershell/files/9414536/CloudShell-VersionTable.txt)
[Local-VersionTable.txt](https://github.com/Azure/azure-powershell/files/9414539/Local-VersionTable.txt)

### Issue script & Debug output

```PowerShell
DEBUG: Registering Az shared AssemblyLoadContext for path: 'C:\Users\bando\Documents\PowerShell\Modules\Az.Accounts\2.9.1\StartupScripts\..\AzSharedAlcAssemblies'.
DEBUG: AssemblyLoadContext registered.

NAME
    New-AzEventGridSubscription

SYNOPSIS
    Creates a new Azure Event Grid Event Subscription to a topic, Azure resource, Azure subscription or Resource Group.


SYNTAX
    New-AzEventGridSubscription [-EventSubscriptionName] <System.String> [-Endpoint] <System.String>
    [[-ResourceGroupName] <System.String>] [-AdvancedFilter <System.Collections.Hashtable[]>]
    [-AzureActiveDirectoryApplicationIdOrUri <System.String>] [-AzureActiveDirectoryTenantId <System.String>]
    [-DeadLetterEndpoint <System.String>] [-DefaultProfile
    <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>] [-DeliverySchema
    {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema |
    CloudEventSchemaV1_0}] [-EndpointType {webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction | webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate <System.DateTime>]
    [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt <System.Int32>]
    [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>] [-SubjectBeginsWith
    <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm] [-WhatIf]
    [<CommonParameters>]

    New-AzEventGridSubscription [-EventSubscriptionName] <System.String> [-ResourceId] <System.String> [-Endpoint]
    <System.String> [-AdvancedFilter <System.Collections.Hashtable[]>] [-AzureActiveDirectoryApplicationIdOrUri
    <System.String>] [-AzureActiveDirectoryTenantId <System.String>] [-DeadLetterEndpoint <System.String>]
    [-DefaultProfile <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>]
    [-DeliverySchema {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema
    | CloudEventSchemaV1_0}] [-EndpointType {webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction | webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate <System.DateTime>]
    [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt <System.Int32>]
    [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>] [-SubjectBeginsWith
    <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm] [-WhatIf]
    [<CommonParameters>]

    New-AzEventGridSubscription [-EventSubscriptionName] <System.String> [-Endpoint] <System.String>
    [-ResourceGroupName] <System.String> [-TopicName] <System.String> [-AdvancedFilter
    <System.Collections.Hashtable[]>] [-AzureActiveDirectoryApplicationIdOrUri <System.String>]
    [-AzureActiveDirectoryTenantId <System.String>] [-DeadLetterEndpoint <System.String>] [-DefaultProfile
    <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>] [-DeliverySchema
    {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema |
    CloudEventSchemaV1_0}] [-EndpointType {webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction | webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate <System.DateTime>]
    [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt <System.Int32>]
    [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>] [-SubjectBeginsWith
    <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm] [-WhatIf]
    [<CommonParameters>]

    New-AzEventGridSubscription [-EventSubscriptionName] <System.String> [-Endpoint] <System.String>
    [-ResourceGroupName] <System.String> [-DomainName] <System.String> [-AdvancedFilter
    <System.Collections.Hashtable[]>] [-AzureActiveDirectoryApplicationIdOrUri <System.String>]
    [-AzureActiveDirectoryTenantId <System.String>] [-DeadLetterEndpoint <System.String>] [-DefaultProfile
    <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>] [-DeliverySchema
    {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema |
    CloudEventSchemaV1_0}] [-EndpointType {webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction | webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate <System.DateTime>]
    [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt <System.Int32>]
    [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>] [-SubjectBeginsWith
    <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm] [-WhatIf]
    [<CommonParameters>]

    New-AzEventGridSubscription [-EventSubscriptionName] <System.String> [-Endpoint] <System.String>
    [-ResourceGroupName] <System.String> [-DomainName] <System.String> [-AdvancedFilter
    <System.Collections.Hashtable[]>] [-AzureActiveDirectoryApplicationIdOrUri <System.String>]
    [-AzureActiveDirectoryTenantId <System.String>] [-DeadLetterEndpoint <System.String>] [-DefaultProfile
    <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>] [-DeliverySchema
    {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema |
    CloudEventSchemaV1_0}] -DomainTopicName <System.String> [-EndpointType {webhook | eventhub | storagequeue |
    hybridconnection | servicebusqueue | servicebustopic | azurefunction | webhook | eventhub | storagequeue |
    hybridconnection | servicebusqueue | servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate
    <System.DateTime>] [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt
    <System.Int32>] [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>]
    [-SubjectBeginsWith <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm]
    [-WhatIf] [<CommonParameters>]

    New-AzEventGridSubscription [-EventSubscriptionName] <System.String> [-InputObject]
    <Microsoft.Azure.Commands.EventGrid.Models.PSTopic> [-Endpoint] <System.String> [-AdvancedFilter
    <System.Collections.Hashtable[]>] [-AzureActiveDirectoryApplicationIdOrUri <System.String>]
    [-AzureActiveDirectoryTenantId <System.String>] [-DeadLetterEndpoint <System.String>] [-DefaultProfile
    <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>] [-DeliverySchema
    {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema |
    CloudEventSchemaV1_0}] [-EndpointType {webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction | webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate <System.DateTime>]
    [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt <System.Int32>]
    [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>] [-SubjectBeginsWith
    <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm] [-WhatIf]
    [<CommonParameters>]

    New-AzEventGridSubscription [-DomainInputObject] <Microsoft.Azure.Commands.EventGrid.Models.PSDomain>
    [-EventSubscriptionName] <System.String> [-Endpoint] <System.String> [-AdvancedFilter
    <System.Collections.Hashtable[]>] [-AzureActiveDirectoryApplicationIdOrUri <System.String>]
    [-AzureActiveDirectoryTenantId <System.String>] [-DeadLetterEndpoint <System.String>] [-DefaultProfile
    <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>] [-DeliverySchema
    {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema |
    CloudEventSchemaV1_0}] [-EndpointType {webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction | webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate <System.DateTime>]
    [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt <System.Int32>]
    [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>] [-SubjectBeginsWith
    <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm] [-WhatIf]
    [<CommonParameters>]

    New-AzEventGridSubscription [-DomainTopicInputObject] <Microsoft.Azure.Commands.EventGrid.Models.PSDomainTopic>
    [-EventSubscriptionName] <System.String> [-Endpoint] <System.String> [-AdvancedFilter
    <System.Collections.Hashtable[]>] [-AzureActiveDirectoryApplicationIdOrUri <System.String>]
    [-AzureActiveDirectoryTenantId <System.String>] [-DeadLetterEndpoint <System.String>] [-DefaultProfile
    <Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer>] [-DeliverySchema
    {EventGridSchema | CustomInputSchema | CloudEventSchemaV1_0 | EventGridSchema | CustomInputSchema |
    CloudEventSchemaV1_0}] [-EndpointType {webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction | webhook | eventhub | storagequeue | hybridconnection | servicebusqueue |
    servicebustopic | azurefunction}] [-EventTtl <System.Int32>] [-ExpirationDate <System.DateTime>]
    [-IncludedEventType <System.String[]>] [-Label <System.String[]>] [-MaxDeliveryAttempt <System.Int32>]
    [-MaxEventsPerBatch <System.Int32>] [-PreferredBatchSizeInKiloBytes <System.Int32>] [-SubjectBeginsWith
    <System.String>] [-SubjectCaseSensitive] [-SubjectEndsWith <System.String>] [-Confirm] [-WhatIf]
    [<CommonParameters>]


DESCRIPTION
    Create a new event subscription to an Azure Event Grid topic, a supported Azure resource, an Azure subscription or
    Resource Group. To create an event subscription to the currently selected Azure subscription, specify the event
    subscription name and the destination endpoint. To create an event subscription to a resource group, specify the
    resource group name in addition to the event subscription name and the destination endpoint. To create an event
    subscription to an Azure Event Grid topic, specify the topic name as well. To create an event subscription to a
    supported Azure resource, specify the full resource ID of the resource. To view the list of supported types, run
    the Get-AzEventGridTopicType cmdlet.


RELATED LINKS
    Online Version: https://docs.microsoft.com/en-us/powershell/module/az.eventgrid/new-azeventgridsubscription

REMARKS
    To see the examples, type: "Get-Help New-AzEventGridSubscription -Examples"
    For more information, type: "Get-Help New-AzEventGridSubscription -Detailed"
    For technical information, type: "Get-Help New-AzEventGridSubscription -Full"
    For online help, type: "Get-Help New-AzEventGridSubscription -Online"
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.2.6
PSEdition                      Core
GitCommitId                    7.2.6
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
Directory: C:\Users\bando\Documents\PowerShell\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     2.9.1                 Az.Accounts                         Core,Desk {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosave…}
Script     1.1.2                 Az.Advisor                          Core,Desk {Get-AzAdvisorRecommendation, Enable-AzAdvisorRecommendation, Disable-AzAdvisorRecommendation, Get-AzAdvisorConfiguration…}
Script     4.2.1                 Az.Aks                              Core,Desk {Get-AzAksCluster, New-AzAksCluster, Remove-AzAksCluster, Import-AzAksCredential…}
Script     1.1.4                 Az.AnalysisServices                 Core,Desk {Resume-AzAnalysisServicesServer, Suspend-AzAnalysisServicesServer, Get-AzAnalysisServicesServer, Remove-AzAnalysisServicesServer…}
Script     3.0.0                 Az.ApiManagement                    Core,Desk {Add-AzApiManagementApiToGateway, Add-AzApiManagementApiToProduct, Add-AzApiManagementProductToGroup, Add-AzApiManagementRegion…}
Script     1.1.0                 Az.AppConfiguration                 Core,Desk {Get-AzAppConfigurationStore, Get-AzAppConfigurationStoreKey, New-AzAppConfigurationStore, New-AzAppConfigurationStoreKey…}
Script     2.1.0                 Az.ApplicationInsights              Core,Desk {Get-AzApplicationInsights, Get-AzApplicationInsightsApiKey, Get-AzApplicationInsightsContinuousExport, Get-AzApplicationInsightsLinkedStorageAccount…}
Script     1.0.0                 Az.Attestation                      Core,Desk {New-AzAttestation, Get-AzAttestation, Remove-AzAttestation, Get-AzAttestationPolicy…}
Script     1.7.3                 Az.Automation                       Core,Desk {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybridWorkerGroup, Get-AzAutomationJobOutputRecord, Import-AzAutomationDscNodeConfiguration…}
Script     3.2.1                 Az.Batch                            Core,Desk {Remove-AzBatchAccount, Get-AzBatchAccount, Get-AzBatchAccountKey, New-AzBatchAccount…}
Script     2.0.0                 Az.Billing                          Core,Desk {Get-AzBillingInvoice, Get-AzBillingPeriod, Get-AzEnrollmentAccount, Get-AzConsumptionBudget…}
Script     2.1.0                 Az.Cdn                              Core,Desk {Clear-AzCdnEndpointContent, Clear-AzFrontDoorCdnEndpointContent, Disable-AzCdnCustomDomainCustomHttps, Enable-AzCdnCustomDomainCustomHttps…}
Script     1.1.0                 Az.CloudService                     Core,Desk {Get-AzCloudService, Get-AzCloudServiceInstanceView, Get-AzCloudServiceNetworkInterface, Get-AzCloudServiceOSFamily…}
Script     1.11.0                Az.CognitiveServices                Core,Desk {Get-AzCognitiveServicesAccount, Get-AzCognitiveServicesAccountKey, Get-AzCognitiveServicesAccountSku, Get-AzCognitiveServicesAccountType…}
Script     4.30.0                Az.Compute                          Core,Desk {Remove-AzAvailabilitySet, Get-AzAvailabilitySet, New-AzAvailabilitySet, Update-AzAvailabilitySet…}
Script     1.0.0                 Az.ConfidentialLedger               Core,Desk {Get-AzConfidentialLedger, New-AzConfidentialLedger, New-AzConfidentialLedgerAADBasedSecurityPrincipalObject, New-AzConfidentialLedgerCertBasedSecurityPrinci…
Script     3.1.0                 Az.ContainerInstance                Core,Desk {Add-AzContainerInstanceOutput, Get-AzContainerGroup, Get-AzContainerInstanceCachedImage, Get-AzContainerInstanceCapability…}
Script     3.0.0                 Az.ContainerRegistry                Core,Desk {New-AzContainerRegistry, Get-AzContainerRegistry, Update-AzContainerRegistry, Remove-AzContainerRegistry…}
Script     1.8.2                 Az.CosmosDB                         Core,Desk {Get-AzCosmosDBSqlContainer, Get-AzCosmosDBSqlContainerThroughput, Get-AzCosmosDBSqlDatabase, Get-AzCosmosDBSqlDatabaseThroughput…}
Script     1.1.0                 Az.DataBoxEdge                      Core,Desk {Get-AzDataBoxEdgeJob, Get-AzDataBoxEdgeDevice, Invoke-AzDataBoxEdgeDevice, New-AzDataBoxEdgeDevice…}
Script     1.2.0                 Az.Databricks                       Core,Desk {Get-AzDatabricksOutboundNetworkDependenciesEndpoint, Get-AzDatabricksVNetPeering, Get-AzDatabricksWorkspace, New-AzDatabricksVNetPeering…}
Script     1.16.8                Az.DataFactory                      Core,Desk {Set-AzDataFactoryV2, Update-AzDataFactoryV2, Get-AzDataFactoryV2, Remove-AzDataFactoryV2…}
Script     1.0.2                 Az.DataLakeAnalytics                Core,Desk {Get-AzDataLakeAnalyticsDataSource, New-AzDataLakeAnalyticsCatalogCredential, Remove-AzDataLakeAnalyticsCatalogCredential, Set-AzDataLakeAnalyticsCatalogCred…
Script     1.3.0                 Az.DataLakeStore                    Core,Desk {Get-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreFirewallRule, Set-AzDataLakeStoreTrustedIdProvider…}
Script     1.0.1                 Az.DataShare                        Core,Desk {New-AzDataShareAccount, Get-AzDataShareAccount, Remove-AzDataShareAccount, New-AzDataShare…}
Script     1.1.0                 Az.DeploymentManager                Core,Desk {Get-AzDeploymentManagerArtifactSource, New-AzDeploymentManagerArtifactSource, Set-AzDeploymentManagerArtifactSource, Remove-AzDeploymentManagerArtifactSourc…
Script     3.1.0                 Az.DesktopVirtualization            Core,Desk {Disconnect-AzWvdUserSession, Expand-AzWvdMsixImage, Get-AzWvdApplication, Get-AzWvdApplicationGroup…}
Script     1.0.2                 Az.DevTestLabs                      Core,Desk {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShutdownPolicy, Get-AzDtlAutoStartPolicy, Get-AzDtlVMsPerLabPolicy…}
Script     1.1.2                 Az.Dns                              Core,Desk {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remove-AzDnsRecordSet, Set-AzDnsRecordSet…}
Script     1.3.0                 Az.EventGrid                        Core,Desk {New-AzEventGridTopic, Get-AzEventGridTopic, Set-AzEventGridTopic, New-AzEventGridTopicKey…}
Script     2.2.0                 Az.EventHub                         Core,Desk {New-AzEventHubNamespace, Get-AzEventHubNamespace, Set-AzEventHubNamespace, Remove-AzEventHubNamespace…}
Script     1.9.0                 Az.FrontDoor                        Core,Desk {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDoor, Remove-AzFrontDoor…}
Script     4.0.3                 Az.Functions                        Core,Desk {Get-AzFunctionApp, Get-AzFunctionAppAvailableLocation, Get-AzFunctionAppPlan, Get-AzFunctionAppSetting…}
Script     5.0.1                 Az.HDInsight                        Core,Desk {Get-AzHDInsightJob, New-AzHDInsightSqoopJobDefinition, Wait-AzHDInsightJob, New-AzHDInsightStreamingMapReduceJobDefinition…}
Script     2.0.0                 Az.HealthcareApis                   Core,Desk {Get-AzHealthcareApisService, Get-AzHealthcareApisWorkspace, Get-AzHealthcareDicomService, Get-AzHealthcareFhirDestination…}
Script     2.7.4                 Az.IotHub                           Core,Desk {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGroup, Get-AzIotHubConnectionString, Get-AzIotHubJob…}
Script     4.6.1                 Az.KeyVault                         Core,Desk {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, Stop-AzKeyVaultCertificateOperation, Get-AzKeyVaultCertificateOperation…}
Script     2.1.0                 Az.Kusto                            Core,Desk {Add-AzKustoClusterLanguageExtension, Add-AzKustoDatabasePrincipal, Get-AzKustoAttachedDatabaseConfiguration, Get-AzKustoCluster…}
Script     1.5.0                 Az.LogicApp                         Core,Desk {Get-AzIntegrationAccountAgreement, Get-AzIntegrationAccountAssembly, Get-AzIntegrationAccountBatchConfiguration, Get-AzIntegrationAccountCallbackUrl…}
Script     1.1.3                 Az.MachineLearning                  Core,Desk {Move-AzMlCommitmentAssociation, Get-AzMlCommitmentAssociation, Get-AzMlCommitmentPlanUsageHistory, Remove-AzMlCommitmentPlan…}
Script     1.2.0                 Az.Maintenance                      Core,Desk {Get-AzApplyUpdate, Get-AzConfigurationAssignment, Get-AzMaintenanceConfiguration, Get-AzMaintenanceUpdate…}
Script     1.0.0                 Az.ManagedServiceIdentity           Core,Desk {Get-AzSystemAssignedIdentity, Get-AzUserAssignedIdentity, New-AzUserAssignedIdentity, Remove-AzUserAssignedIdentity…}
Script     3.0.0                 Az.ManagedServices                  Core,Desk {Get-AzManagedServicesAssignment, Get-AzManagedServicesDefinition, Get-AzManagedServicesMarketplaceDefinition, New-AzManagedServicesAssignment…}
Script     1.0.2                 Az.MarketplaceOrdering              Core,Desk {Get-AzMarketplaceTerms, Set-AzMarketplaceTerms}
Script     1.1.1                 Az.Media                            Core,Desk {Sync-AzMediaServiceStorageKey, Set-AzMediaServiceKey, Get-AzMediaServiceKey, Get-AzMediaServiceNameAvailability…}
Script     1.1.3                 Az.Migrate                          Core,Desk {Get-AzMigrateDiscoveredServer, Get-AzMigrateJob, Get-AzMigrateProject, Get-AzMigrateReplicationEligibilityResult…}
Script     3.0.1                 Az.Monitor                          Core,Desk {Get-AzMetricDefinition, Get-AzMetric, Remove-AzLogProfile, Get-AzLogProfile…}
Script     1.0.0                 Az.MySql                            Core,Desk {Get-AzMySqlConfiguration, Get-AzMySqlConnectionString, Get-AzMySqlFirewallRule, Get-AzMySqlFlexibleServer…}
Script     4.20.0                Az.Network                          Core,Desk {Add-AzApplicationGatewayAuthenticationCertificate, Get-AzApplicationGatewayAuthenticationCertificate, New-AzApplicationGatewayAuthenticationCertificate, Rem…
Script     1.1.1                 Az.NotificationHubs                 Core,Desk {Get-AzNotificationHub, Get-AzNotificationHubAuthorizationRule, Get-AzNotificationHubListKey, Get-AzNotificationHubPNSCredential…}
Script     3.1.0                 Az.OperationalInsights              Core,Desk {New-AzOperationalInsightsAzureActivityLogDataSource, New-AzOperationalInsightsCustomLogDataSource, Disable-AzOperationalInsightsLinuxCustomLogCollection, Di…
Script     1.5.0                 Az.PolicyInsights                   Core,Desk {Get-AzPolicyEvent, Get-AzPolicyState, Get-AzPolicyStateSummary, Get-AzPolicyRemediation…}
Script     1.1.0                 Az.PostgreSql                       Core,Desk {Get-AzPostgreSqlConfiguration, Get-AzPostgreSqlConnectionString, Get-AzPostgreSqlFirewallRule, Get-AzPostgreSqlFlexibleServer…}
Script     1.2.0                 Az.PowerBIEmbedded                  Core,Desk {Remove-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollectionAccessKey, Get-AzPowerBIWorkspace…}
Script     1.0.3                 Az.PrivateDns                       Core,Desk {Get-AzPrivateDnsZone, Remove-AzPrivateDnsZone, Set-AzPrivateDnsZone, New-AzPrivateDnsZone…}
Script     5.5.0                 Az.RecoveryServices                 Core,Desk {Get-AzRecoveryServicesBackupProperty, Get-AzRecoveryServicesVault, Get-AzRecoveryServicesVaultSettingsFile, New-AzRecoveryServicesVault…}
Script     1.6.0                 Az.RedisCache                       Core,Desk {Remove-AzRedisCachePatchSchedule, New-AzRedisCacheScheduleEntry, Get-AzRedisCachePatchSchedule, New-AzRedisCachePatchSchedule…}
Script     1.1.0                 Az.RedisEnterpriseCache             Core,Desk {Export-AzRedisEnterpriseCache, Get-AzRedisEnterpriseCache, Get-AzRedisEnterpriseCacheDatabase, Get-AzRedisEnterpriseCacheKey…}
Script     1.0.3                 Az.Relay                            Core,Desk {New-AzRelayNamespace, Get-AzRelayNamespace, Set-AzRelayNamespace, Remove-AzRelayNamespace…}
Script     1.1.0                 Az.ResourceMover                    Core,Desk {Add-AzResourceMoverMoveResource, Get-AzResourceMoverMoveCollection, Get-AzResourceMoverMoveResource, Get-AzResourceMoverRequiredForResources…}
Script     6.1.0                 Az.Resources                        Core,Desk {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignment…}
Script     1.3.0                 Az.Security                         Core,Desk {Get-AzSecuritySolution, Get-AzSecuritySolutionsReferenceData, New-AzAlertsSuppressionRuleScope, Get-AzAlertsSuppressionRule…}
Script     1.1.0                 Az.SecurityInsights                 Core,Desk {Get-AzSentinelAlertRuleAction, New-AzSentinelAlertRuleAction, Remove-AzSentinelAlertRuleAction, Update-AzSentinelAlertRuleAction…}
Script     1.10.0                Az.ServiceBus                       Core,Desk {New-AzServiceBusNamespace, Get-AzServiceBusNamespace, Set-AzServiceBusNamespace, Remove-AzServiceBusNamespace…}
Script     3.1.0                 Az.ServiceFabric                    Core,Desk {Add-AzServiceFabricClientCertificate, Add-AzServiceFabricNode, Add-AzServiceFabricNodeType, Get-AzServiceFabricCluster…}
Script     1.4.1                 Az.SignalR                          Core,Desk {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey, New-AzSignalRKey…}
Script     3.11.0                Az.Sql                              Core,Desk {Get-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseTransparentDataEncryptionActivity, Set-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabas…
Script     1.1.0                 Az.SqlVirtualMachine                Core,Desk {New-AzSqlVM, Get-AzSqlVM, Update-AzSqlVM, Remove-AzSqlVM…}
Script     1.3.0                 Az.StackHCI                         Core,Desk {Add-AzStackHCIVMAttestation, Disable-AzStackHCIAttestation, Disable-AzStackHCIRemoteSupport, Enable-AzStackHCIAttestation…}
Script     4.8.0                 Az.Storage                          Core,Desk {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStorageAccount, New-AzStorageAccountKey…}
Script     1.7.0                 Az.StorageSync                      Core,Desk {Invoke-AzStorageSyncCompatibilityCheck, New-AzStorageSyncService, Get-AzStorageSyncService, Set-AzStorageSyncService…}
Script     2.0.0                 Az.StreamAnalytics                  Core,Desk {Get-AzStreamAnalyticsCluster, Get-AzStreamAnalyticsClusterStreamingJob, Get-AzStreamAnalyticsDefaultFunctionDefinition, Get-AzStreamAnalyticsFunction…}
Script     1.0.0                 Az.Support                          Core,Desk {Get-AzSupportService, Get-AzSupportProblemClassification, Get-AzSupportTicket, Get-AzSupportTicketCommunication…}
Script     1.5.0                 Az.Synapse                          Core,Desk {Get-AzSynapseSparkJob, Stop-AzSynapseSparkJob, Submit-AzSynapseSparkJob, Wait-AzSynapseSparkJob…}
Script     1.1.0                 Az.TrafficManager                   Core,Desk {Add-AzTrafficManagerCustomHeaderToEndpoint, Remove-AzTrafficManagerCustomHeaderFromEndpoint, Add-AzTrafficManagerCustomHeaderToProfile, Remove-AzTrafficMana…
Script     2.11.3                Az.Websites                         Core,Desk {Get-AzAppServicePlan, Set-AzAppServicePlan, New-AzAppServicePlan, Remove-AzAppServicePlan…}
```

**Comment by dingmeng-xue** [comment]
Thanks for reporting. There are couple possibilities. Please double check the version of module. Besides that, PowerShell allows cmdlet to expose parameter according to OS/Platform. It needs service team to confirm it. 

If it is true, cmdlet developer should add more information into reference doc.

**Comment by bandolir** [comment]
Thanks. Here are the loaded modules on both platform.

[CloudShell-GetModuleOutput.txt](https://github.com/Azure/azure-powershell/files/9422273/CloudShell-GetModuleOutput.txt)
[local-GetModuleOutput.txt](https://github.com/Azure/azure-powershell/files/9422274/local-GetModuleOutput.txt)

**Comment by dingmeng-xue** [comment]
Thanks for reporting. The root cause is due to https://github.com/Azure/azure-powershell/issues/19299#issuecomment-1227210805. This module will be updated at the beginning of September.

**Comment by bandolir** [comment]
Thanks.

**Comment by navba-MSFT** [comment]
@bandolir Thanks for reaching out to us. Please let us know if you have any follow-up queries on this.

**Comment by dingmeng-xue** [comment]
Thanks @navba-MSFT for following up. I believe this difference should be fixed in the next week after Az 8.3 release. I'm closing this issue now.
